app/services/dish_search_service.rb

```diff
@@ -0,0 +1,86 @@
+class DishSearchService
+  attr_accessor :search_params 
+
+  def initialize(search_params = {})
+    @search_params = parse_params search_params
+  end
+
+  def search
+    @dishes = Dish.search search_params[:q], 
+      page: search_params[:page], 
+      per_page: search_params[:per_page],
+      order: order,
+      where: where,
+      facets: [:food_preference, :food_type, :cuisine]
+  end
+
+  # serialized search query string
+  def search_string
+    "q=#{@search_params[:q]}&" + 
+    "sort=#{@search_params[:sort]}&" + 
+    "per_page=#{@search_params[:per_page]}&" + 
+    "lat=#{@search_params[:lat]}&" + 
+    "lon=#{@search_params[:lon]}&" +
+    "within_dist=#{@search_params[:within_dist]}"
+  end
+
+  def geocoded?
+    true if @search_params[:lat] and @search_params[:lon]
```

Comment:
Just realized that some code is using "lng" and other code is using "lon". It'd be good to standardize this if possible.